keras_cv/models/segmentation/deeplabv3plus.py

```diff
@@ -0,0 +1,319 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+from keras_cv.layers.spatial_pyramid import SpatialPyramidPooling
+from keras_cv.models import utils
+from keras_cv.models.weights import parse_weights
+
+
+@keras.utils.register_keras_serializable(package="keras_cv")
+class DeepLabV3Plus(keras.Model):
+    """
+    A segmentation model based on the DeepLabV3Plus model.
+
+    Args:
+        classes: int, the number of classes for the detection model. Note that
+            the classes doesn't contain the background class, and the classes
+            from the data should be represented by integers with range
+            [0, classes).
+        backbone: a backbone for the model, expected to be a KerasCV model.
+            Typically ResNet50V2 or ResNet101V2. Default 'low_level_feature_layer' assumes
+            either and uses 'v2_stack_1_block4_1_relu' for them by default.
+        low_level_feature_layer: the layer name for the low-level features to use for encoding/decoding
```

Comment:
preferrably this should be `backbone_level`, you can check what `Resnet50V2.as_backbone` returns

Reply:
Originally wanted to as well - but that prevents us from using non-ResNet backbones, such as Xception (which is proposed in the paper), since no other model has `as_backbone()`.

I could make `backbone_level` available if the backbone is a ResNet, and not available otherwise?

Reply:
We should consider adding `as_backbone` for all other models?

Reply:
At least for the common backbones, like ResNets. In my experience, EffNets didn't prove to be competitive against ResNets for backbones like these (possibly due to the SE blocks?)

Anyhow, it seems like an okay compromise to use ResNet's `as_backbone()` now, block the argument for other architectures and then gradually expand the number of models that supports `as_backbone()` as we see use for them?